pkg/controller/deployment/sync.go

```diff
@@ -300,6 +300,12 @@ func (dc *DeploymentController) scale(deployment *apps.Deployment, newRS *apps.R
 	// deployment. If there is no active replica set, then we should scale up the newest replica set.
 	if activeOrLatest := deploymentutil.FindActiveOrLatest(newRS, oldRSs); activeOrLatest != nil {
 		if *(activeOrLatest.Spec.Replicas) == *(deployment.Spec.Replicas) {
+			annotationsNeedUpdate := deploymentutil.ReplicasAnnotationsNeedUpdate(activeOrLatest, *(deployment.Spec.Replicas), *(deployment.Spec.Replicas)+deploymentutil.MaxSurge(*deployment))
```

Comment:
Can we add a test to cover this?

Reply:
Add a new test case for this, @wzshiming please take a look.